src/Evolution/DgSubcell/Actions/TciAndRollback.hpp

```diff
@@ -0,0 +1,252 @@
+            // G+D, which is what we would be projecting here.
+          });
+
+      if (UNLIKELY(db::get<::Tags::TimeStepId>(box).slab_number() < 0)) {
+        // If we are doing self start, then we need to project the initial guess
+        // to the subcells as well.
+        //
+        // Warning: this unfortunately needs to be kept in sync with the
+        //          self-start procedure.
+        //
+        // Note: if we switch to the subcells then we might have an inconsistent
+        //       state between the primitive and conservative variables on the
+        //       subcells. The most correct thing is to re-compute the primitive
+        //       variables on the subcells, since projecting the conservative
+        //       variables is conservative.
+        if constexpr (Metavariables::system::
+                          has_primitive_and_conservative_vars) {
+          db::mutate<
+              SelfStart::Tags::InitialValue<variables_tag>,
+              SelfStart::Tags::InitialValue<
+                  typename Metavariables::system::primitive_variables_tag>>(
+              make_not_null(&box),
+              [&dg_mesh, &subcell_mesh](
+                  const auto initial_vars_ptr,
+                  const auto initial_prim_vars_ptr) noexcept {
+                // Note: for strict conservation, we need to project uJ instead
+                // of just u.
+                std::get<0>(*initial_vars_ptr) =
+                    fd::project(std::get<0>(*initial_vars_ptr), dg_mesh,
+                                subcell_mesh.extents());
+                std::get<0>(*initial_prim_vars_ptr) =
+                    fd::project(std::get<0>(*initial_prim_vars_ptr), dg_mesh,
+                                subcell_mesh.extents());
+              });
+        } else {
+          db::mutate<SelfStart::Tags::InitialValue<variables_tag>>(
+              make_not_null(&box),
+              [&dg_mesh, &subcell_mesh](const auto initial_vars_ptr) noexcept {
+                // Note: for strict conservation, we need to project uJ instead
+                // of just u.
+                std::get<0>(*initial_vars_ptr) =
+                    fd::project(std::get<0>(*initial_vars_ptr), dg_mesh,
+                                subcell_mesh.extents());
+              });
+        }
+      }
+      // We could/should project the _old_ primitive variables to the subcells
+      // to get a good initial guess.
+
+      return {std::move(box), false,
+              tmpl::index_of<
+                  ActionList,
+                  ::Actions::Label<evolution::dg::subcell::Actions::Labels::
+                                       BeginSubcellAfterDgRollback>>::value +
+                  1};
+    }
+    // The unlimited DG solver has passed, so we can remove the current neighbor
+    // data.
+    db::mutate<subcell::Tags::NeighborDataForReconstructionAndRdmpTci<Dim>>(
+        make_not_null(&box), [](const auto nhbr_data_ptr) noexcept {
```

Comment:
[style nitpick, feel free to ignore] I would slightly prefer 'nhbr' -> 'neighbor' throughout the PR (i.e. here and in the test file) because it's easier to read and not much longer